src/main/java/org/assertj/core/internal/Objects.java

```diff
@@ -651,6 +653,23 @@ private boolean isActualIn(Object actual, Iterable<?> values) {
   }
 
   /**
+   * Assert that the given object is "deeply" equals to the other by comparing all fields recursively.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given object.
+   * @param other the object to compare {@code actual} to.
+   * @throws AssertionError if actual is {@code null}.
+   * @throws AssertionError if the other object is not an instance of the actual type.
+   * @throws AssertionError if the actual and the given object are not "deeply" equal.
+   */
+  public <A> void assertIsEqualToComparingFieldByFieldRecursively(AssertionInfo info, A actual, A other) {
+    assertNotNull(info, actual);
+    assertHasSameClassAs(info, actual, other);
```

Comment:
remove the class check, this constraint being sometimes too strong - see [this guy complaining](https://github.com/joel-costigliola/assertj-core/issues/412) ;-)